src/main/323C.c

```diff
@@ -1491,15 +1508,42 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B468);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B560);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B694);
+void func_8002B694(struct Unk* arg0)
+{
+    *(s32*)&arg0->x_pos_hi += arg0->unk20;
+    *(s32*)&arg0->y_pos_hi -= arg0->unk24;
+
+    if (arg0->unk15) {
+        arg0->unk20 += arg0->unk28;
+    } else {
+        arg0->unk20 -= arg0->unk28;
+    }
+
+    arg0->unk24 -= arg0->unk2C;
+    if (arg0->unk24 < FIXED(-6.5)) {
+        arg0->unk24 = FIXED(-6.5);
```

Comment:
Good sign that some of these fields are using fixed-point types.

Reply:
Seems like we should go with x_pos, y_pos as the f32 type?
https://decomp.me/scratch/LCm6C
https://decomp.me/scratch/zemvr

Reply:
Yup, makes sense! Do you want to do that as part of this PR or a follow-up? (feel free to push commits to this branch, or I can look in a bit)

Reply:
It doesn't have to be done right now, the pattern was becoming clear so figured I'd mention it

Reply:
Easy enough to do that cleanup now. I also had hi and lo reversed, so this fixes that.